e2e/cmd/keys.go

```diff
@@ -0,0 +1,252 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"reflect"
+
+	"github.com/omni-network/omni/e2e/app"
+	"github.com/omni-network/omni/e2e/app/eoa"
+	"github.com/omni-network/omni/e2e/app/key"
+	"github.com/omni-network/omni/e2e/types"
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/log"
+	"github.com/omni-network/omni/lib/netconf"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/naoina/toml"
+	"github.com/spf13/cobra"
+)
+
+func newKeyCreate(def *app.Definition) *cobra.Command {
+	cfg := key.UploadConfig{}
+
+	cmd := &cobra.Command{
+		Use:   "key-create",
+		Short: "Creates a private key in GCP secret manager for a node in a manifest",
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			if def.Testnet.Network == netconf.Simnet || def.Testnet.Network == netconf.Devnet {
+				return errors.New("cannot create keys for simnet or devnet")
+			}
+
+			cfg.Network = def.Testnet.Network
+
+			if err := verifyKeyNodeType(*def, cfg); err != nil {
+				return errors.Wrap(err, "verify cfg", "name", cfg.Name, "type", cfg.Type)
+			}
+
+			_, err := key.UploadNew(cmd.Context(), cfg)
+
+			return err
+		},
+	}
+
+	bindKeyCreateFlags(cmd, &cfg)
+
+	return cmd
+}
+
+func newKeyCreateAll(def *app.Definition) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "key-create-all",
+		Short: "Creates (if required) all private keys in GCP secret manager for nodes in a manifest",
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			if def.Testnet.Network == netconf.Simnet || def.Testnet.Network == netconf.Devnet {
+				return errors.New("cannot create keys for simnet or devnet")
+			}
+
+			ctx := cmd.Context()
+			network := def.Testnet.Network
+			manifest := def.Manifest
+			if manifest.Keys == nil {
+				manifest.Keys = make(map[string]map[key.Type]string)
+			}
+
+			for _, cfg := range allKeyConfigs(*def) {
+				log.Debug(ctx, "Checking key", "name", cfg.Name, "type", cfg.Type)
+
+				// Check if key already defined
+				if cfg.Type == key.EOA {
+					_, ok := eoa.Address(network, eoa.Role(cfg.Name))
+					if ok {
+						log.Info(ctx, "Skipping eoa key already defined", "name", cfg.Name)
```

Comment:
Nit: this is a bit hard to parse without punctuation.
```suggestion
						log.Info(ctx, "Skipping: eoa key already defined", "name", cfg.Name)
```

Reply:
not sure I agree